docs/csharp/fundamentals/program-structure/main-command-line.md

```diff
@@ -105,64 +108,81 @@ Execution succeeded
 Return value = 0
 ```
 
-### Async Main return values
+### Main return Task
+
+When the application entry point returns a `Task` or `Task<int>`, the compiler generates a new entry point that calls the entry point method declared in the application code. Assuming that this entry point is called `$GeneratedMain`, the compiler generates the following code for these entry points:
+
+- `static Task Main()` results in the compiler emitting the equivalent of `private static void $GeneratedMain() => Main().GetAwaiter().GetResult();`
+- `static Task Main(string[])` results in the compiler emitting the equivalent of `private static void $GeneratedMain(string[] args) => Main(args).GetAwaiter().GetResult();`
+- `static Task<int> Main()` results in the compiler emitting the equivalent of `private static int $GeneratedMain() => Main().GetAwaiter().GetResult();`
+- `static Task<int> Main(string[])` results in the compiler emitting the equivalent of `private static int $GeneratedMain(string[] args) => Main(args).GetAwaiter().GetResult();`
```

Comment:
I've remove all these lines.

Reply:
The thing to keep in mind is to document:

- The model
- The surface area
- The contract

This is what we do. `$GeneratedMain` is none of those things.

FWIW ... I've been down this path many times. I find something that is interesting and meaningful to me and then start to document it (in a blog post or docs). The team then reminds me that it isn't _what_ we delivered but _how_ we delivered it. I then see their viewpoint and delete that content for official sites (like docs).

However, this can be interesting content for your own blog post, video, or other medium if that is of interest to you.

Reply:
I consider to rewrite this part in another PR.
I created the issue #41457 to follow up.